scripts/find_equation_id.rb

```diff
@@ -0,0 +1,649 @@
+#!/usr/bin/env ruby
+
+=begin
+This module maps magma equations from/to their id
+
+It can be used as a script in interactive mode (with the -i switch), as
+
+    ruby find_equation_id.rb -i
+
+or by passing arguments to it from stdin or as arguments: a (space-separated)
+list of ids or of equations (in which the operation can be ".", "*", or "◇"),
+optionally preceeded by "*" to dualize the equation (and characters "[,]" are
+ignored):
+
+    ruby find_equation_id.rb [12, 34] "(w*u)=t*(u*x)" 4567 "*89" "*67" "0=(1*2)*(0*1)"
+
+When used as a module imported in python code, one can use
```

Comment:
Should this be changed to ruby or deleted? I'm not sure if the API matches the python code here 100%